Source/Core/VideoBackends/Vulkan/VKTexture.cpp

```diff
@@ -0,0 +1,339 @@
+// Copyright 2009 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include <algorithm>
+#include <cstddef>
+#include <cstring>
+
+#include "Common/Align.h"
+#include "Common/Assert.h"
+#include "Common/CommonTypes.h"
+#include "Common/Logging/Log.h"
+#include "Common/MsgHandler.h"
+
+#include "VideoBackends/Vulkan/CommandBufferManager.h"
+#include "VideoBackends/Vulkan/FramebufferManager.h"
+#include "VideoBackends/Vulkan/StagingTexture2D.h"
+#include "VideoBackends/Vulkan/StateTracker.h"
+#include "VideoBackends/Vulkan/Texture2D.h"
+#include "VideoBackends/Vulkan/Util.h"
+#include "VideoBackends/Vulkan/VKTexture.h"
+#include "VideoBackends/Vulkan/VulkanContext.h"
+
+#include "VideoCommon/ImageWrite.h"
+#include "VideoCommon/TextureConfig.h"
+
+namespace Vulkan
+{
+VKTexture::VKTexture(const TextureConfig& tex_config, std::unique_ptr<Texture2D> texture,
+                     VkFramebuffer framebuffer)
+    : AbstractTexture(tex_config), m_framebuffer(framebuffer), m_texture(std::move(texture))
+{
+}
+
+VKTexture* VKTexture::Create(const TextureConfig& tex_config)
```

Comment:
This gets passed to a  unique_ptr in Vulkan's TextureCache::CreateTexture (assuming it is non-null).  This is to get around the fact there's no return type on a constructor.  I wanted to use exceptions but eh.  The unique_ptr in TextureCache::CreateTexture is of type AbstractTexture.  I didn't see of a good way to "cast" the unique_ptr<VKTexture> to unique_ptr<AbstractTexture> so I opted for a raw ptr here instead.

Please let me know if you have any ideas.